### ISSUE TITLE
Improve consistency in parameter names

### ISSUE THREAD

**Issue by bellet** [scikit-learn-contrib/metric-learn#257]
It would be nice to follow a consistent naming convention for parameters and be as consistent as possible with sklearn. For instance:
- In supervised versions of weakly supervised algorithms, `num_constraints` should be renamed `n_constraints`, `num_chunks` to `n_chunks`
- In LMNN, the parameter `k` could be renamed `n_neighbors` like in sklearn's `KNeighborsClassifier`
- There is also `tol` and `convergence_threshold` which are both used to refer to optimization tolerance (we should always use `tol` which is quite standard, cf `scipy.optimize)
